tests/test_geoserver.py

```diff
@@ -46,17 +63,90 @@ def get_geoserver_settings():
+    """
+    Prepares a workspace, its datastore and a test shapefile along with the associated Geoserver resources.
+    """
+    # Creating workspace and datastore needed to publish a shapefile
+    workspace_name = test_instance.workspaces[workspace_key]
+    datastore_name = f"test-datastore-{workspace_key}"
+    geoserver_workspace_path = f"/user_workspaces/{workspace_name}/shapefile_datastore"
+    geoserver_handler._create_workspace_request(workspace_name=workspace_name)
+    geoserver_handler._create_datastore_request(workspace_name=workspace_name,
+                                                datastore_name=datastore_name)
+    geoserver_handler._configure_datastore_request(workspace_name=workspace_name,
+                                                   datastore_name=datastore_name,
+                                                   datastore_path=geoserver_workspace_path)
+
+    # Preparations needed to make tests work without all the other handlers running
+    datastore_path = get_datastore_path(test_instance.workspace_folders[workspace_key])
+
+    # This next part can fail if the user running this test doesn't have write access to the directory
+    copy_shapefile(basename=test_instance.test_shapefile_name, destination=datastore_path)
+
+    return workspace_name, datastore_name
+
+
+def reset_geoserver_test_workspace(test_instance, geoserver_handler):
+    """
+    Removes a workspace on the file system and the associated resources on Geoserver.
+    """
+    for _, workspace in test_instance.workspaces.items():
+        try:
+            geoserver_handler._remove_workspace_request(workspace_name=workspace)
+        except GeoserverError:
+            # Making sure all test workspaces are removed
+            pass
+    for _, folder in test_instance.workspace_folders.items():
+        try:
+            if not folder.startswith("/tmp/"):
+                raise PermissionError("Aborting test workspace reset. The test workspace path should be in the `/tmp`"
+                                      f"directory, but was instead found at the path `{folder}`.")
+            # Make sure access permissions are enabled before deleting files
+            os.chmod(folder, 0o777)
+            for root, dirs, files in os.walk(folder):
+                for resource in dirs + files:
+                    os.chmod(os.path.join(root, resource), 0o777)
+            shutil.rmtree(folder)
+        except FileNotFoundError:
+            pass
+
+
 def copy_shapefile(basename, destination):
     full_filename = f"{COWBIRD_ROOT}/tests/resources/{basename}"
     Path(destination).mkdir(parents=True, exist_ok=False)
     for file in glob.glob(f"{full_filename}.*"):
         shutil.copy(file, destination)
 
 
-@pytest.mark.online
-@pytest.mark.geoserver
-class TestGeoserverRequests:
+def get_datastore_path(workspace_path):
+    return workspace_path + "/shapefile_datastore"
+
+
+class TestGeoserver:
```

Comment:
Why remove the markers? Allows running a subset of tests.

Reply:
I have removed the markers on the generic `TestGeoserver` class (that has no tests) and added them to the 2 subclasses `TestGeoserverRequests` and `TestGeoserverPermissions`, that have tests. We can still fully run a subset of tests using the markers on the subclasses.